pkg/collector/corechecks/net/wlan/wlan_darwin.m

```diff
@@ -0,0 +1,73 @@
+#import <Foundation/Foundation.h>
+#import <CoreLocation/CoreLocation.h>
+#import <CoreWLAN/CoreWLAN.h>
+#import "wlan_darwin.h"
+
+@interface LocationManager : NSObject <CLLocationManagerDelegate>
+
+@property (nonatomic, strong) CLLocationManager *locationManager;
+@property (nonatomic, assign) BOOL isAuthorized;
+
+- (instancetype)init;
+- (BOOL)checkLocationPermissions;
+
+@end
+
+@implementation LocationManager
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        self.locationManager = [[CLLocationManager alloc] init];
+        self.locationManager.delegate = self;
+        self.isAuthorized = NO;
+
+        if (@available(macOS 10.15, *)) {
```

Comment:
Do we care about supporting < mac os 10.15?

Reply:
As far as I know, we don't. Its most recent update was in July 20, 2022.

Reply:
OK - only asking because as it is, `isAuthorized` will always be false on old mac os version. Just wanted to make sure this was the intended default.